corehq/apps/style/static/style/js/pagination.ng.js

```diff
@@ -0,0 +1,121 @@
+(function (angular, undefined) {
+    'use strict';
+
+    var pagination = angular.module('hq.pagination', [
+        'ui.bootstrap',
+        'ngResource',
+        'ngRoute',
+        'ngCookies',
+        'ng.django.rmi'
+    ]);
+
+    pagination.constant('paginationLimits', [
+        [10, "Limit 10"]
+    ]);
+
+    pagination.constant('paginationLimitCookieName', 'hq.ngPaginationLimit');
+
+    pagination.constant("paginationCustomData", {});
+
+    var paginationControllers = {};
+
+    paginationControllers.PaginatedListController = function(
+        $scope, djangoRMI, paginationConfig, paginationLimits,
+        paginationCustomData, paginationLimitCookieName, $cookies
+    ) {
+        var self = this;
+        $scope._ = _;  // makes underscore available
+        $scope.djangoRMI = djangoRMI;
+        $scope.paginatedItems = [];
+
+        $scope.paginationLimits = _.map(paginationLimits, function (l) {
+            return {
+                value: l[0],
+                key: l[1]
+            };
+        });
+
+        var storedLimit = $cookies.get(paginationLimitCookieName);
+        if (_.isString(storedLimit)) {
+            try {
+                storedLimit = parseInt(storedLimit);
+            } catch (e) {
+                // do nothing
+            }
+        }
+        if (!_.isNumber(storedLimit)) {
+            storedLimit = 10;
+        }
+        $scope.limit = storedLimit;
+
+        $scope.total = 1;
+        $scope.currentPage = 1;
+
+        $scope.query = '';
+
+        $scope.notLoaded = true;
+        $scope.hasError = false;
+
+        self.retries = 0;
+
+        $scope.paginationCustomData = paginationCustomData;
+        _.each(paginationCustomData, function (val, key) {
+            $scope[key] = val;
+        });
```

Comment:
Since these keys are user-defined, there's some chance for name conflicts (however small) - should this be namespaced?  Something like `$scope.customData[key] = val;`